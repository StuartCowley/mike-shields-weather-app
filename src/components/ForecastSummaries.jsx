import React from "react";
import PropTypes from "prop-types";
import styles from "../styles/ForecastSummaries.module.css";
import ForecastSummary from "./ForecastSummary";

export default function ForecastSummaries({ forecasts, handleForecastSelect }) {
  return (
    <div
      className={styles["forecast-summaries"]}
      data-testid="forecast-summaries"
    >
      {forecasts.map((forecast) => {
        return (
          <ForecastSummary
            key={`forecast-summary-${date}`}
            date={date}
            description={description}
            temperature={temperature}
            icon={icon}
          />
        );
      })}
    </div>
  );
}

ForecastSummaries.propTypes = {
  forecasts: PropTypes.arrayOf(
    PropTypes.shape({
      date: PropTypes.number,
      description: PropTypes.string,
      icon: PropTypes.string,
      temperature: PropTypes.shape({
        max: PropTypes.number,
        min: PropTypes.number,
      }),
    })
  ).isRequired,
};
