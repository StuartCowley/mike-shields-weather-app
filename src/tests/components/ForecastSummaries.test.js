import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";
import forecast from "../../data/forecast.json";

const { forecasts } = forecast;

describe("ForecastSummaries", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(
      <ForecastSummaries
        forecasts={forecasts}
        handleForecastSelect={jest.fn()}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders forecasts as ForecastSummary components", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries forecasts={forecasts} handleForecastSelect={jest.fn()}/>
    );
    expect(getAllByTestId("forecast-summary")).toHaveLength(forecasts.length);
  });
});
