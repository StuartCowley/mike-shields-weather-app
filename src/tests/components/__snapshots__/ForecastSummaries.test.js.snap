// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
    data-testid="forecast-summaries"
  >
    <div
      class="forecast-summary shadow"
      data-testid="forecast-summary"
    >
      <h2
        class="forecast-summary__date"
        data-testid="forecast-summary-date"
      >
        <span>
          Mon 30th Apr
        </span>
      </h2>
      <p
        class="forecast-summary__description"
        data-testid="forecast-summary-description"
      >
        Clear
      </p>
      <div
        class="weather-icon-component"
        data-testid="weather-icon-component"
      >
        <i
          class="wi wi-day-sunny"
          name="owm"
          style="transform: scale(2.75); color: rgb(255, 255, 255);"
        />
      </div>
      <p
        class="forecast-summary__temperature"
        data-testid="forecast-summary-temperature"
      >
        <span>
          11
        </span>
        °C
      </p>
      <button
        class="shadow"
        type="button"
      >
        More details...
      </button>
    </div>
    <div
      class="forecast-summary shadow"
      data-testid="forecast-summary"
    >
      <h2
        class="forecast-summary__date"
        data-testid="forecast-summary-date"
      >
        <span>
          Tue 1st May
        </span>
      </h2>
      <p
        class="forecast-summary__description"
        data-testid="forecast-summary-description"
      >
        Stormy
      </p>
      <div
        class="weather-icon-component"
        data-testid="weather-icon-component"
      >
        <i
          class="wi wi-day-thunderstorm"
          name="owm"
          style="transform: scale(2.75); color: rgb(255, 255, 255);"
        />
      </div>
      <p
        class="forecast-summary__temperature"
        data-testid="forecast-summary-temperature"
      >
        <span>
          13
        </span>
        °C
      </p>
      <button
        class="shadow"
        type="button"
      >
        More details...
      </button>
    </div>
    <div
      class="forecast-summary shadow"
      data-testid="forecast-summary"
    >
      <h2
        class="forecast-summary__date"
        data-testid="forecast-summary-date"
      >
        <span>
          Wed 2nd May
        </span>
      </h2>
      <p
        class="forecast-summary__description"
        data-testid="forecast-summary-description"
      >
        Heavy Snow
      </p>
      <div
        class="weather-icon-component"
        data-testid="weather-icon-component"
      >
        <i
          class="wi wi-day-snow"
          name="owm"
          style="transform: scale(2.75); color: rgb(255, 255, 255);"
        />
      </div>
      <p
        class="forecast-summary__temperature"
        data-testid="forecast-summary-temperature"
      >
        <span>
          1
        </span>
        °C
      </p>
      <button
        class="shadow"
        type="button"
      >
        More details...
      </button>
    </div>
    <div
      class="forecast-summary shadow"
      data-testid="forecast-summary"
    >
      <h2
        class="forecast-summary__date"
        data-testid="forecast-summary-date"
      >
        <span>
          Thu 3rd May
        </span>
      </h2>
      <p
        class="forecast-summary__description"
        data-testid="forecast-summary-description"
      >
        Tornado
      </p>
      <div
        class="weather-icon-component"
        data-testid="weather-icon-component"
      >
        <i
          class="wi wi-tornado"
          name="owm"
          style="transform: scale(2.75); color: rgb(255, 255, 255);"
        />
      </div>
      <p
        class="forecast-summary__temperature"
        data-testid="forecast-summary-temperature"
      >
        <span>
          20
        </span>
        °C
      </p>
      <button
        class="shadow"
        type="button"
      >
        More details...
      </button>
    </div>
    <div
      class="forecast-summary shadow"
      data-testid="forecast-summary"
    >
      <h2
        class="forecast-summary__date"
        data-testid="forecast-summary-date"
      >
        <span>
          Fri 4th May
        </span>
      </h2>
      <p
        class="forecast-summary__description"
        data-testid="forecast-summary-description"
      >
        Hazy
      </p>
      <div
        class="weather-icon-component"
        data-testid="weather-icon-component"
      >
        <i
          class="wi wi-day-haze"
          name="owm"
          style="transform: scale(2.75); color: rgb(255, 255, 255);"
        />
      </div>
      <p
        class="forecast-summary__temperature"
        data-testid="forecast-summary-temperature"
      >
        <span>
          25
        </span>
        °C
      </p>
      <button
        class="shadow"
        type="button"
      >
        More details...
      </button>
    </div>
  </div>
</DocumentFragment>
`;
